File: leetcode/s0701_0800/leetcode_746.py
from typing import *


class leetcode_746(object):
    """leetcode_746 documentation"""

    def __init__(self, *args, **kwargs):
        """Constructor for leetcode_746"""
        pass

    def solve(self):
        pass

    def minCostClimbingStairs(self, cost: List[int]) -> int:
        a = [cost[0], cost[1]]
        for i in range(2, len(cost)):
            a.append(min([a[i-1]+cost[i], a[i-2]+cost[i]]))
        return min([a[len(a)-1], a[len(a)-2]])


if __name__ == '__main__':
    l: leetcode_746 = leetcode_746()
    l.solve()

File: src/leetcode/s0701_800/leetcode_746.java
package leetcode.s0701_800;

public class leetcode_746 {
    public int minCostClimbingStairs(int[] cost) {
        int f1=0, f2=0;
        for (int i=cost.length-1;i>=0;i--) {
            int f0 = cost[i] + Math.min(f1, f2);
            f2 = f1;
            f1=f0;
        }
        return Math.min(f1,f2);
    }

    public static void main(String[] args) {
        leetcode_746 t = new leetcode_746();
        System.out.println(t.minCostClimbingStairs(new int[]{0, 0, 1, 1}));
    }
}
